plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -378,8 +382,29 @@ function MosaicMenuItem:init()
             style = self.shortcut_style,
         }
     end
+
+    -- self.progress_bar = ProgressWidget:new
+
+
     self.detail = self.text
 
+    if not self.menu.cover_info_cache then
+        self.menu.cover_info_cache = {}
+    end
+
+    self.progress = nil
+    self.percentage, self.has_cover = unpack(self:getProgress())
```

Comment:
I would prefer to have all this done "in the flow" in MosaicMenuItem:update(), like how it is done in ListMenuItem:update(). So the logic for both stays consistent.
So no getProgress() factorisation/externalisation needed.
It's indeed a bit chunk of code, but it's somehow optimized to do stuff only in the branches that shows it is needed.

Reply:
I've followed the pattern of `corner_mark` this time.